game/src/common/waypoints.rs

```diff
@@ -221,8 +218,42 @@ impl InputWaypoints {
 
     fn update_waypoints_drawable(&mut self, ctx: &mut EventCtx) {
         let mut batch = GeomBatch::new();
-        for waypt in &self.waypoints {
-            batch.append(waypt.geom.clone());
+        let total_waypoints = self.waypoints.len();
+        for (idx, waypt) in &mut self.waypoints.iter().enumerate() {
+            let geom = {
+                let wp = {
```

Comment:
Since we only use the relative position here to pick color, maybe a full enum and two steps to pick the color is overkill? And I'm not sure the pattern matching does much for us here. What about just
```
let color = if idx == 0 {
  Color::GREEN
  } else if idx == total_waypoints - 1 {
   Color::RED
   } else {
[Color::BLUE, Color::ORANGE, Color::PURPLE][idx % 3]
};
```

Reply:
Updated.